Session4/ArrOperation.java

```diff
@@ -0,0 +1,185 @@
+import java.security.acl.LastOwnerException;
+import java.util.ArrayList;
+
+/**
+ * Return the size of the largest mirror section found in the input array.
+ * Return the number of clumps in the input array.
+ * Solve fixXY problem
+ * Return the index if there is a place to split the input array 
+ *
+ * @author Aman Gautam 
+ * 
+ * Dated 18/07/2019
+ */
+public class ArrOperation{
+	int maxMirror(int mirrorArray[]) {
+
+		if (mirrorArray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+
+		int maxLength = 0;
+		int tempLength = 0;
+		int pivot;
+		for (int indexI = 0; indexI < mirrorArray.length; indexI++) {
+			pivot = indexI;
+			tempLength = 0;
+			for (int indexJ = mirrorArray.length - 1; indexJ >= indexI; indexJ--) {
+				if (mirrorArray[pivot] == mirrorArray[indexJ]) {
+					tempLength++;
+					pivot++;
+				} else if (tempLength > maxLength) {
+					maxLength = tempLength;
+					tempLength = 0;
+					pivot = indexI;
+				} else if(pivot!=indexI){
+					tempLength = 0;
+					pivot = indexI;
+					indexJ++;
+				}
+				if (tempLength > maxLength)
+					maxLength = tempLength;
+			}
+		}
+		return maxLength;
+	}
+	
+	/**Return the number of clumps in the input array.
+	 * 
+	 * @param inputarray[]
+	 * @return clumps
+	 */
+	 static int noofClumps(int[] inputarray) {
+		int clumps = 0, LastIndex;
+		boolean CheckingStatus;
+
+		if (inputarray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+		for (int FirstIndex = 0; FirstIndex < inputarray.length; FirstIndex++) {
+			CheckingStatus = false;
+			for (LastIndex = 1; (LastIndex + FirstIndex) < inputarray.length; LastIndex++) {
+				if (inputarray[FirstIndex] == inputarray[LastIndex + FirstIndex]) {
+					CheckingStatus = true;
+				} else
+					break;
+			}
+			if (CheckingStatus) {
+				clumps++;
+				FirstIndex = FirstIndex + LastIndex - 1;
+			}
+		}
+		System.out.println(clumps);
+		return clumps;
+	}
+
+	
+	static boolean checkadjecency(int[] inputarray ,int X) {
+		for(int count=0;count<inputarray.length-1;count++)
+		{
+			if(inputarray[count]==X && inputarray[count+1]==X) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	static int countOfX(int[] inputArray, int X) {
+		int countX = 0;
+		for (int count = 0; count < inputArray.length; count++) {
+			if (inputArray[count] == X)
+				countX++;
+		}
+		return countX;
+	}
+	
```

Comment:
why two methods, can use one method for both the purpose of counting.

Reply:
changed to a single method countOf_XorY